File: Python/Hill_German.py
from codecs import open
from PreProcess_German import PreProcess_German
from ASCII_Enc_German import ASCII_Enc_German
from ASCII_Dec_German import ASCII_Dec_German
from ASCII_30_Add import ASCII_30_Add
from ASCII_30_Del import ASCII_30_Del
from EncryptDecryptHill_German import EncryptDecryptHill_German
from Generate import Generate

# Hill cipher for the German Alphabet

file = open("nachtlied.txt", 'r', 'iso-8859-1')
rawtext = file.read()
file.close()

plaintext = PreProcess_German(rawtext)
plaintext_ascii = list(map(ord, plaintext))
plaintext_ascii_30 = ASCII_30_Add(plaintext_ascii)

[key, inverted_key] = Generate(8, 30)

ciphertext_ascii_30 = EncryptDecryptHill_German(
    [(i - 97) for i in plaintext_ascii_30], key, 30)
ciphertext_ascii = ASCII_Enc_German([(i + 65) for i in ciphertext_ascii_30])

dectext_ascii_30 = EncryptDecryptHill_German(ciphertext_ascii_30, inverted_key, 30)
dectext_ascii = ASCII_30_Del([(i + 97) for i in dectext_ascii_30])

ciphertext = ''.join(map(chr, ciphertext_ascii))
dectext = ''.join(map(chr, dectext_ascii))

print(plaintext[300:400])
print(ciphertext[300:400])
print(dectext[300:400])

File: MATLAB/Hill_German.m
% Hill cipher for the German Alphabet
f=fopen('nachtlied.txt','r');
a=fread(f);
fclose(f);
x=PreProcess_German(a');
x1 = ASCII_30_Add(x);

[K K_1 i]=Generate(8,30);
mod(K*K_1,30);
mod(K_1*K,30);
y=EncryptDecryptHill_German(x1-97,K,30);
y1 = ASCII_Enc_German(y(300:400) +65);
y2 = ASCII_Dec_German(y1);
z=EncryptDecryptHill_German(y,K_1,30);
z1 = ASCII_30_Del(z(300:400) +97);

cat(1,char(x(300:400)),char(y1),char(z1))